### PR TITLE
Snapshot restore applies cloud init change

### PR DESCRIPTION
**Pull request by georgeliao**
This PR is about applying the extra interface changes to cloud-init-config.iso file when the snapshot retore command is called.

### CODE REVIEW

File: include/multipass/virtual_machine.h [hunk 1]
```diff
@@ -85,8 +85,8 @@ class VirtualMachine : private DisabledCopyMove
     virtual void resize_memory(const MemorySize& new_size) = 0;
     virtual void resize_disk(const MemorySize& new_size) = 0;
     virtual void add_network_interface(int index, const NetworkInterface& net) = 0;
-    virtual void add_extra_interfaces_to_cloud_init(const std::string& default_mac_addr,
-                                                    const std::vector<NetworkInterface>& extra_interfaces) = 0;
+    virtual void apply_extra_interfaces_to_cloud_init(const std::string& default_mac_addr,
+                                                      const std::vector<NetworkInterface>& extra_interfaces) = 0;
```
Comment (georgeliao) [inline include/multipass/virtual_machine.h:89]: Name prefix changed from add to apply which better reflects what the function is doing.

File: src/platform/backends/shared/base_virtual_machine.cpp [hunk 2]
```diff
@@ -541,6 +541,7 @@ void BaseVirtualMachine::restore_snapshot(const std::string& name, VMSpecs& spec
     specs.num_cores = snapshot->get_num_cores();
     specs.mem_size = snapshot->get_mem_size();
     specs.disk_space = snapshot->get_disk_space();
+    const bool is_extra_interfaces_different = specs.extra_interfaces != snapshot->get_extra_interfaces();
```
Comment (luis4a0) [inline src/platform/backends/shared/base_virtual_machine.cpp:544]: I'd change this to `are_extra_interfaces_different`. I know it's a data structure, but it sounds better to me to have the plural.

**Review by luis4a0 — CHANGES_REQUESTED** [review]
Hey @georgeliao! This looks good. I've made a minor comment inline about a variable name change. Then, I'd like you to rebase on the current code of `cloud-init-extra-interfaces-config_plus_snapshot`, so we have the current code OK. Finally, I have to make some more functional tests, but everything seems OK. Thanks for this!

File: src/platform/backends/shared/base_virtual_machine.cpp (hunk 2 above)
Comment (georgeliao) [reply src/platform/backends/shared/base_virtual_machine.cpp:544]: Yep, that is a better name. Will do.

**Review by luis4a0 — APPROVED** [review]
Alright @georgeliao, this works as intended, does what it is supposed to so we are ready to merge. Thanks for this!

**Pull request merged**